docs/rest_api.md

```diff
@@ -1,253 +1,289 @@
+* nic - The nic which should be used to connect the node to provisioning network
+* channel - The vlan that should be used to connect the node to provisioning network
 
 The convention that we are following for requests is:
+* PUT - for resource creation like node creation 
+* DELETE - for resource deletion for node deletion
+* POST - for rest of operations
 
-PUT - for resource creation like node creation 
-DELETE - for resource deletion for node deletion
-POST - for rest of operations
+**The username and password for HaaS needs to be passed along using HTTP Basic Auth to each possible API call.**
 
 Each possible API call has:
+* an HTTP method and URL path
+* Request body(which will always be form encoded parameters)
+* A list of possible responses
+* An example
 
--an HTTP method and URL path
--Request body(which will always be a JSON object)
--A list of possible responses
--An example
-
-Provision:
-
+---
+###Provision:
 Provision API is needed for provisioning a node from MOC cluster as of now. Provision operation internally calls a ceph clone operation. Ceph clone operation usually takes a snap_
 
 Following is the call for API:
 
-http://BMI_SERVER:PORT/provision_node 
-with PUT request type and request body as:
+####Link:
+http://BMI_SERVER:PORT/provision/
+
+####Request Type:
+PUT
 
+####Request Body:
+```json
 {
-"node" : "<node_name>" , 
-"img" : "<image_name>" , 
-"project" : "<project_name>" , 
-"snap_name" : "<snapshot_name>"
-} 
-
-Responses:
-
-- 200. This means the provision call is successful.
-- Internal 500 with some junk characters. This means the request body is not proper.
-- 401. This means unauthorized access to ceph image or snapshot or image already exists in ceph.
-- 403. This means a ceph connection problem.
-- 409. Image busy exception.
-- 444. You used a wrong request method like PUT instead of POST etc.
+ "project" : "<project_name>",
+ "node" : "<node_name>" , 
+ "img" : "<image_name>" ,
+ "network" : "<network_name>" ,
+ "channel" : "<channel in haas>" ,
+ "nic" : "<nic to connect on>"
+}
+```
+
+####Responses:
+* 200. This means the provision call is successful.
+* Internal 500 with some junk characters. This means the request body is not proper.
+* 401. This means unauthorized access to ceph image or snapshot or image already exists in ceph.
+* 403. This means a ceph connection problem.
+* 409. Image busy exception.
+* 444. You used a wrong request method like PUT instead of POST etc.
  
-Example:
-
-https://<BMI_SERVER>:<PORT>/provision_node
-with PUT request type and request body as:
-
-body:
+####Example:
+Send a PUT Request with following body to http://<BMI_SERVER>:<PORT>/provision/
 
+```json
 {
- "node" : "cisco-2016" ,
- "img" : "hadoopMaster.img" ,
- "project" : "bmi_infra" ,
- "snap_name" : "HadoopMasterGoldenImage"
+ "project" : "bmi_infra",
+ "node" : "cisco-2016" , 
+ "img" : "hadoopMaster.img" ,  
+ "network" : "provision-net" ,
+ "channel" : "vlan/native",
```

Comment:
Do we need to specify channel? Shouldn't it be always native as long as we use HIL.